UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -584,26 +590,35 @@ void UpdateSensors()
         /// </summary>
         public void GenerateSensorData()
         {
-
             int floatsWritten = 0;
             // Generate data for all Sensors
             for (var i = 0; i < sensors.Count; i++)
             {
                 var sensor = sensors[i];
                 if (sensor.GetCompressionType() == SensorCompressionType.None)
                 {
-                    m_WriteAdapter.SetTarget(m_Info.floatObservations, floatsWritten);
-                    floatsWritten += sensor.Write(m_WriteAdapter);
+                    // only handles 1D
```

Comment:
There's currently an assumption that compressed observation <=> 3 dimensional. We should (later) add an option to write visual obs as floats, just to test all the code paths (and maybe a compressed vector obs too).